exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetPredicatesHelper.java

```diff
@@ -39,22 +40,21 @@ static boolean isNullOrEmpty(Statistics stat) {
    *
    * @param stat parquet column statistics
    * @param rowCount number of rows in the parquet file
-   * @return True if all rows are null in the parquet file
-   *          False if at least one row is not null.
+   * @return <tt>true</tt> if all rows are null in the parquet file and <tt>false</tt> otherwise
    */
   static boolean isAllNulls(Statistics stat, long rowCount) {
-    return stat.isNumNullsSet() && stat.getNumNulls() == rowCount;
+    Preconditions.checkArgument(rowCount >= 0, String.format("negative rowCount %d is not valid", rowCount));
```

Comment:
Why do you want to fail in this case? Maybe just warning?

Reply:
When can rowCount be negative?

Reply:
I don't know, why you have added this check then? :)

Reply:
To avoid bugs :). It is invalid to call `isAllNulls()` with negative `rowCount`.

Reply:
I think this check is not needed here :) Why this method should check what was given to it? It's job just to compare row count and number of nulls.

Reply:
To validate input. It can't give the correct answer if an input is a junk (`rowCount` is negative).

Reply:
Ok, then just please describe what will happen during query execution if negative row comes into this method, will query fail or statistics will be ignored and there will not filter push down? I hope for the second.

Reply:
I hope for the reverse. Negative row count passed to this method indicates a bug, and bug means incorrect result. I'd rather fail a query than give a wrong result.

Reply:
But this negative rowCount could only come from the statistics, I don't think that bad statistics should prevent user from executing the query.

Reply:
It does not matter where it comes from (it actually comes from `RowGroupInfo`). The condition needs to be intercepted prior to calling `isAllNulls()` as `isAllNulls()` would return the wrong result (whether true or false) for a negative row count.